### PR TITLE
Throw only specialized package-only exceptions

### PR DESCRIPTION
**Pull request by Majkl578**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

#### Summary

Stop throwing around SPL exceptions, only throw our own.

BC is kept through inheritance.

### CODE REVIEW

File: lib/Doctrine/Migrations/Finder/Exception/FInderException.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Finder\Exception;
+
+interface FInderException
```
Comment (greg0ire) [inline lib/Doctrine/Migrations/Finder/Exception/FInderException.php]: case issue

File: lib/Doctrine/Migrations/Finder/Exception/NameIsReserved.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+use function sprintf;
+
+final class NameIsReserved extends InvalidArgumentException implements FInderException
+{
+    public static function new(string $version) : self
+    {
+        return new self(sprintf(
+            'Cannot load a migrations with the name "%s" because it is a reserved number by Doctrine Migrations.'
+            . PHP_EOL
+            . "It's used to revert all migrations including the first one.",
```
Comment (greg0ire) [inline lib/Doctrine/Migrations/Finder/Exception/NameIsReserved.php]: `It is`, and simple quotes?

Comment (Majkl578) [reply lib/Doctrine/Migrations/Finder/Exception/NameIsReserved.php]: @greg0ire Some of the messages would indeed benefit from rewording, mind to send a PR? ;)

Comment (greg0ire) [reply lib/Doctrine/Migrations/Finder/Exception/NameIsReserved.php]: Will do!

**Pull request merged**